reports/tests/test_views.py

```diff
@@ -1,7 +1,266 @@
-# Could not get this work yet:
-# - "axes.exceptions.AxesBackendRequestParameterRequired:
-# AxesBackend requires a request as an argument to authenticate"
-#
-# def test_an_admin_view(admin_client):
-#     response = admin_client.get("/admin/")
-#     assert response.status_code == 200
+import datetime
+from unittest import mock
+
+import pytest
+from django.test import override_settings, TestCase
+from django.test.client import RequestFactory
+from django.utils import timezone
+from django.views.generic import TemplateView
+from graphql_relay.node.node import to_global_id
+from occurrences.factories import (
+    EnrolmentFactory,
+    OccurrenceFactory,
+    PalvelutarjotinEventFactory,
+    StudyGroupFactory,
+)
+from occurrences.models import Enrolment, PalvelutarjotinEvent
+from organisations.factories import OrganisationFactory, PersonFactory
+from organisations.models import Organisation
+from reports.views import (
+    ExportReportViewMixin,
+    OrganisationPersonsAdminView,
+    OrganisationPersonsMixin,
+    PalvelutarjotinEventEnrolmentsAdminView,
+    PalvelutarjotinEventEnrolmentsMixin,
+)
+
+
+class ExportReportViewMixinTest(TestCase):
+    class DummyExportReportView(ExportReportViewMixin, TemplateView):
+        model = PalvelutarjotinEvent
+
+    def setUp(self):
+        super(ExportReportViewMixinTest, self).setUp()
+
+        # Setup request and view.
+        self.view = self.DummyExportReportView()
+
+    @pytest.mark.django_db
+    def test_get_queryset(self):
+        (
+            p_event1,
+            p_event2,
+            p_event3,
+            p_event4,
+        ) = PalvelutarjotinEventFactory.create_batch(4)
+        self.view.request = RequestFactory().get(
```

Comment:
I think we can also test the filter here by creating 5 p_events but only query for 4 
You can also use `snapshot` fixture to match the data instead of only relying on the number of p_events return

Reply:
Ain't this quite the same without snapshots? 
```
    def test_get_queryset(self):
        (
            p_event1,
            p_event2,
            p_event3,
            p_event4,
        ) = PalvelutarjotinEventFactory.create_batch(4)
        self.view.request = RequestFactory().get(
            "/fake-path?ids=%s,%s,%s" % (p_event1.id, p_event2.id, p_event3.id)
        )
        self.assertEqual(
            set(self.view.get_queryset()), set([p_event1, p_event2, p_event3])
        )
```